src/app/icd/ICDEventManager.cpp

```diff
@@ -26,8 +26,8 @@ CHIP_ERROR ICDEventManager::Init(ICDManager * icdManager)
 {
     VerifyOrReturnError(icdManager != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
     mICDManager = icdManager;
-
-    PlatformMgr().AddEventHandler(ICDEventHandler, reinterpret_cast<intptr_t>(nullptr));
+    mICDManager->Init();
```

Comment:
would it be better of having the init in the `server.cpp` file before passing it to the ICDEventManager.h?